pkg/instance/repository.go

```diff
@@ -12,22 +14,34 @@ import (
 type Repository interface {
 	Link(firstInstance, secondInstance *model.Instance) error
 	Unlink(instance *model.Instance) error
-	Create(instance *model.Instance) error
 	Save(instance *model.Instance) error
 	FindWithParametersById(id uint) (*model.Instance, error)
 	FindByNameAndGroup(instance string, group string) (*model.Instance, error)
 	SaveDeployLog(instance *model.Instance, log string) error
 	FindById(id uint) (*model.Instance, error)
 	Delete(id uint) error
 	FindByGroupNames(names []string) ([]*model.Instance, error)
+	FindWithDecryptedParametersById(id uint) (*model.Instance, error)
 }
 
 type repository struct {
-	db *gorm.DB
+	db     *gorm.DB
+	config config.Config
```

Comment:
I would only depend on what is needed. This repo does not need the entire config. AFAIK it only needs the `InstanceParameterEncryptionKey` 😄